Engine/Setup/BacktestingSetupHandler.cs

```diff
@@ -190,6 +190,12 @@ public bool Setup(SetupHandlerParameters parameters)
                         algorithm.SetEndDate(job.PeriodFinish.Value);
                     }
 
+                    //set initial cash, if present in the job
+                    if (job.CashAmount.HasValue)
```

Comment:
I believe we need to update `QuantConnect.Queues.JobQueue.NextJob() Line 150` not to send in a cash amount else it will always use it instead of the algorithm cash amount. This can be see when running the regression tests.

We should probably loop through the entire `algorithm.Portfolio.CashBook` and set those amounts to 0, similar to https://github.com/QuantConnect/Lean/blob/master/Engine/Setup/BrokerageSetupHandler.cs#L229 . `algorithm.SetCash()` will default to the account currency and might leave other currencies set.

Reply:
Speaking of regression test, does it make sense to define a regression test for this feature? There's a lot of moving parts between job/setup and algorithm execution.